servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java

```diff
@@ -956,13 +959,27 @@ void loadObjectStore() {
         zmsConfig.setUserAuthority(userAuthority);
 
         objectStore = objFactory.create(keyStore);
-        dbService = new DBService(objectStore, auditLogger, zmsConfig, auditReferenceValidator);
+        dbService = new DBService(objectStore, auditLogger, zmsConfig, auditReferenceValidator, authHistoryStore);
 
         // create our group fetcher based on the db service
 
         groupMemberFetcher = new ZMSGroupMembersFetcher(dbService);
     }
 
+    void loadAuthHistoryStore() {
+        String authHistoryFactoryClass = System.getProperty(ZMSConsts.ZMS_PROP_AUTH_HISTORY_STORE_FACTORY_CLASS,
+                ZMSConsts.ZMS_AUTH_HISTORY_STORE_FACTORY_CLASS);
+        AuthHistoryStoreFactory authHistoryStoreFactory;
+        try {
+            authHistoryStoreFactory = (AuthHistoryStoreFactory) Class.forName(authHistoryFactoryClass).newInstance();
+        } catch (InstantiationException | IllegalAccessException | ClassNotFoundException e) {
+            LOG.error("Invalid AuthHistoryStoreFactory class: {} error: {}", authHistoryFactoryClass, e.getMessage());
```

Comment:
let's start just including the exception as the last argument so the full backtrace is logged instead of just the message. when the server fails sometimes these error messages are not that useful and having the full stack is better.